scripts/generate.js

```diff
@@ -0,0 +1,144 @@
+      return -1;
+    }
+    if (a[field] > b[field]) {
+      return 1;
+    }
+    return 0;
+  };
+};
+
+const getEntityModels = () => {
+  return axios(EntityModelUrl)
+    .then((res) => res.data)
+    .then((result) => {
+      return Object.values(result)
+        .filter(Boolean)
+        .filter(entity => entity.bullhornEntitySchema && entity.bullhornEntitySchema.restAccess && entity.bullhornEntitySchema.restAccess.indexOf("READ") > -1)
+        .sort(sortByField("entityName"));
+    });
+};
+
+const generate = async (config = {}) => {
+  try {
+    let dir = config.directory || "./typings";
+    Logger.log(`retrieving metadata for each entity...`);
+    let models = await getEntityModels();
+    Logger.log(`parsing entity models...`);
+    const types = models.map((model) => parseModelData(model));
+
+    Logger.log(`writing templates...`);
+    let deletables = types.filter(
+      (i) => i && i.properties.find((p) => p.name === "isDeleted")
+    );
+    let tmp = CORE_DEFINITION_TEMPLATE({
+      types: types,
+      deletables: deletables,
+    });
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir, { recursive: true });
+    }
+    fs.writeFile(`${dir}/index.ts`, tmp, (err) => {
+      if (err) {
+        Logger.error(err.message);
+      }
+      Logger.success(`Done!`);
+    });
+  } catch (err) {
+    Logger.error("✗ An error occurred", err);
+  }
+};
+
+const parseModelData = (model) => {
+  if (!model) {
+    Logger.error("Missing entity model");
+    return;
+  }
+
+  Logger.log(`${model.entityName}`);
+  let data = {
+    type: model.entityName,
+    properties: [],
+    dependencies: new Set(),
+    dynamic: false,
+  };
+  let hasCustomObjects = false;
+  if (model.entityName.indexOf("CustomObject") >= 0) {
+    data.dynamic = true;
+  } else {
+    for (let field of Object.values(model.fields).sort(
+      sortByField("fieldName")
+    )) {
+      if (["OneToMany", "ManyToMany"].includes(field.associationType)) {
+        data.properties.push({
+          name: field.fieldName,
+          type: `ToMany<${field.associatedEntityType}>`,
+        });
+        data.dependencies.add(field.associatedEntityType);
+      } else if (["ManyToOne", "OneToOne"].includes(field.associationType)) {
+        data.properties.push({
+          name: field.fieldName,
+          type: `${field.associatedEntityType}`,
+        });
+        data.dependencies.add(field.associatedEntityType);
+      } else if (["id"].indexOf(field.fieldName) < 0) {
+        switch (field.fieldType || "") {
+          case "Integer":
+          case "Double":
+          case "Float":
+          case "BigDecimal":
+          case "Long":
+            data.properties.push({
+              name: field.fieldName,
+              type: "number",
+            });
+            break;
+          case "Date":
+          case "Timestamp":
+            data.properties.push({
+              name: field.fieldName,
+              type: "Date",
```

Comment:
Dates are not actually represented as Date objects in REST - they are returned as the millisecond timestamp, which should be of type `number`, not `Date`.